File: class/practice_10.1/match.cpp
#include "match.hpp"

Match::Match(Team& t1, Team& t2, int g1, int g2, int y1, int y2, int r1, int r2, double h1, double h2)
    : team1(t1), team2(t2), goals_team1(g1), goals_team2(g2), yellow_cards_team1(y1),
      yellow_cards_team2(y2), red_cards_team1(r1), red_cards_team2(r2),
      top_height_team1(h1), top_height_team2(h2) {
    update_teams();
}

void Match::update_teams() {
    team1.update_match_result(goals_team1, goals_team2, yellow_cards_team1, red_cards_team1, top_height_team1);
    team2.update_match_result(goals_team2, goals_team1, yellow_cards_team2, red_cards_team2, top_height_team2);

    if (goals_team1 > goals_team2) {
        team1.points += 3.14;
        team2.points -= 0.5;
    } else if (goals_team1 < goals_team2) {
        team1.points -= 0.5;
        team2.points += 3.14;
    } else {
        team1.points += 2.71828;
        team2.points += 2.71828;
    }
}

File: home/home_10.1/match.swift
import Foundation

class Match {
    var team1: Team
    var team2: Team
    var goalsTeam1: Int
    var goalsTeam2: Int
    var yellowCardsTeam1: Int
    var yellowCardsTeam2: Int
    var redCardsTeam1: Int
    var redCardsTeam2: Int
    var topHeightTeam1: Double
    var topHeightTeam2: Double

    init(team1: Team, team2: Team, goalsTeam1: Int, goalsTeam2: Int, yellowCardsTeam1: Int, yellowCardsTeam2: Int, redCardsTeam1: Int, redCardsTeam2: Int, topHeightTeam1: Double, topHeightTeam2: Double) {
        self.team1 = team1
        self.team2 = team2
        self.goalsTeam1 = goalsTeam1
        self.goalsTeam2 = goalsTeam2
        self.yellowCardsTeam1 = yellowCardsTeam1
        self.yellowCardsTeam2 = yellowCardsTeam2
        self.redCardsTeam1 = redCardsTeam1
        self.redCardsTeam2 = redCardsTeam2
        self.topHeightTeam1 = topHeightTeam1
        self.topHeightTeam2 = topHeightTeam2
        updateTeams()
    }

    func updateTeams() {
        team1.updateMatchResult(scored: goalsTeam1, conceded: goalsTeam2, yellow: yellowCardsTeam1, red: redCardsTeam1, topHeight: topHeightTeam1)
        team2.updateMatchResult(scored: goalsTeam2, conceded: goalsTeam1, yellow: yellowCardsTeam2, red: redCardsTeam2, topHeight: topHeightTeam2)

        if goalsTeam1 > goalsTeam2 {
            team1.points += 3.14
            team2.points -= 0.5
        } else if goalsTeam1 < goalsTeam2 {
            team1.points -= 0.5
            team2.points += 3.14
        } else {
            team1.points += 2.71828
            team2.points += 2.71828
        }
    }
}

